indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java

```diff
@@ -417,7 +472,7 @@ public String getVersion(final Interval interval)
 
             final Committer committer = committerSupplier.get();
             final CountDownLatch persistLatch = new CountDownLatch(1);
-            plumber.persist(
+            driver.persist(
```

Comment:
Segments should be published and handed off as well as being persisted. `driver.publishAndRegisterHandoff()` will do all needed things.

Reply:
Please add a test to check the remaining segments at this point are published well.

Reply:
I'm not sure thats the correct behavior.  From my understanding, this block will only be executed when the task is "stopped gracefully".  In that case, I believe the behavior is to persist the state as is with the assumption that the task will be resumed or restarted.

In the happy path case, the segments will have already been persisted.  If the task exists with an exception, this block won't get executed.

Thoughts?  If my analysis is correct, I can try to make the behavior more explicit in the code

Reply:
Below is the current behavior:
https://github.com/druid-io/druid/blob/80419752b53b1fe27cc59e6e07a0e8e89cfe5869/indexing-service/src/main/java/io/druid/indexing/common/task/RealtimeIndexTask.java#L444

It only calls `plumber.finishJob()` if the task was not "gracefully stopped"

Reply:
Yeah, you're correct. But, why is publishing segments moved from the finally block to the try block? It makes me confused.